src/GraphQL.AspNetCore.GraphQL/GraphQLMiddleware.cs

```diff
@@ -0,0 +1,93 @@
+        ///     Throws <see cref="ArgumentNullException" /> if <paramref name="next" /> or <paramref name="options" /> is null.
+        /// </exception>
+        public GraphQLMiddleware(RequestDelegate next, IOptions<GraphQLOptions> options) {
+            if(next == null) {
+                throw new ArgumentNullException(nameof(next));
+            }
+            if(options == null) {
+                throw new ArgumentNullException(nameof(options));
+            }
+            if(options.Value?.Schema == null) {
+                throw new ArgumentException("Schema is null");
+            }
+
+            this.next = next;
+            var optionsValue = options.Value;
+            graphqlPath = string.IsNullOrEmpty(optionsValue?.GraphQLPath) ? GraphQLOptions.DefaultGraphQLPath : optionsValue.GraphQLPath;
+            schema = optionsValue?.Schema;
+        }
+
+        /// <summary>
+        ///     Invokes the middleware with the specified context.
+        /// </summary>
+        /// <param name="context">
+        ///     The context.
+        /// </param>
+        /// <returns>
+        ///     A <see cref="Task" /> representing the middleware invocation.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        ///     Throws <see cref="ArgumentNullException" /> if <paramref name="context" />.
+        /// </exception>
+        public async Task Invoke(HttpContext context) {
+            if(context == null) {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if(ShouldRespondToRequest(context.Request)) {
+                var executionResult = await ExecuteAsync(context.Request).ConfigureAwait(true);
+                await WriteResponseAsync(context.Response, executionResult).ConfigureAwait(true);
+            }
+
+            await next(context).ConfigureAwait(true);
+        }
+
+        private async Task<ExecutionResult> ExecuteAsync(HttpRequest request) {
+            string requestBodyText;
+            using(var streamReader = new StreamReader(request.Body)) {
+                requestBodyText = await streamReader.ReadToEndAsync().ConfigureAwait(true);
+            }
+            var graphqlRequest = JsonConvert.DeserializeObject<GraphQLRequest>(requestBodyText);
+            return await new DocumentExecuter().ExecuteAsync(schema, null, graphqlRequest.Query, graphqlRequest.OperationName, graphqlRequest.Variables.ToInputs()).ConfigureAwait(true);
+        }
+
+        private bool ShouldRespondToRequest(HttpRequest request) {
+            return string.Equals(request.Method, "POST", StringComparison.OrdinalIgnoreCase) && request.Path.Equals(graphqlPath);
+        }
+
+        private static Task WriteResponseAsync(HttpResponse response, ExecutionResult executionResult) {
+            response.ContentType = "application/json";
+            response.StatusCode = executionResult.Errors?.Count == 0 ? 200 : 400;
```

Comment:
Wouldn't this make having no errors (errors being null) default to showing error 400? That's what I'm getting when I run this locally. The result is working perfectly fine however.

Should switch this to `response.StatusCode = executionResult.Errors?.Count > 0 ? 400 : 200;`